mysqld_exporter.go

```diff
@@ -1643,6 +1666,33 @@ func scrapeTableSchema(db *sql.DB, ch chan<- prometheus.Metric) error {
 	}
 
 	for _, database := range dbList {
+		if _, ok := tableSchemaLastRun[database]; !ok {
+			tableSchemaLastRun[database] = time.Time{}
+		}
+		random := rand.Float64() * 0.25
+		nextRun := *tableSchemaAutoAnalyzeDuration + time.Duration(float64(*tableSchemaAutoAnalyzeDuration)*random)
+		if *tableSchemaAutoAnalyze && time.Since(tableSchemaLastRun[database]) > nextRun {
```

Comment:
In different news, since scrapes could run multiple times in parallel, you need to protect access to `tableSchemaLastRun` by a mutex. Probably best to protect this whole if block.